src/main/java/gui/CameraShotBlock.java

```diff
@@ -24,13 +26,15 @@
      * @param rootCenterArea - the rootCenterArea this shot belongs to
      * @param beginCount - the begin count of this shot
      * @param endCount - the end count of this shot
+     * @param grid - the grid this camerashotblock should belong to
      */
     public CameraShotBlock(int shotId, int timetableNumber, RootCenterArea rootCenterArea,
-                           double beginCount, double endCount) {
+                           double beginCount, double endCount, TimelinesGridPane grid) {
         super(rootCenterArea, beginCount, endCount);
         this.shotId = shotId;
         this.timetableNumber = timetableNumber;
         thisBlock = this;
```

Comment:
Couldn't we just use 'this' wherever the variable thisBlock is being used? Probably has a good reason, but I'm curious.